### PR TITLE
Pathlibify font_manager (only internally, doesn't change the API).

### PR DESCRIPTION
**Pull request by anntzer**
## PR Summary

EDIT: also rely on winreg context-managing, https://docs.python.org/3.4/library/winreg.html#registry-handle-objects

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is PEP 8 compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by QuLogic** [comment]
Everything is failing...

**Comment by anntzer** [comment]
should hopefully work better now...

File: lib/matplotlib/font_manager.py [hunk 1]
```diff
@@ -164,19 +160,20 @@ def get_fontext_synonyms(fontext):
...
-    possibly upper-cased, found recursively under the directory.
+    found recursively under the directory.
     """
-    pattern = ';'.join(['*.%s;*.%s' % (ext, ext.upper())
-                        for ext in extensions])
-    return cbook.listFiles(directory, pattern)
+    return [str(path)
+            for ext in extensions
+            for path in filter(Path.is_file,
+                               Path(directory).rglob("*.{}".format(ext)))]
```
Comment (QuLogic) [inline lib/matplotlib/font_manager.py]: This walks the directory tree as many times as there are extensions instead of just once.

Comment (anntzer) [reply lib/matplotlib/font_manager.py]: fixed

**Pull request merged**